### PR TITLE
Add a transport package with libmono-profiler-aot.so on Android

### PR DESCRIPTION
**Pull request by directhex**
This is partially cursed, as runtime.git's packaging infra is opinionated about e.g. overriding PackageID

### CODE REVIEW

**Review by jonathanpeppers — COMMENTED** [review]
Does this PR produce the package? Was going to download and look at it.

File: src/mono/mono/profiler/CMakeLists.txt [hunk 1]
```diff
@@ -28,6 +28,13 @@ if(NOT DISABLE_LIBS)
     endif()
   endif()
 
+  if(HOST_ANDROID)
+    add_library(mono-profiler-aot SHARED aot.c helper.c)
+    target_compile_definitions(mono-profiler-aot PRIVATE -DMONO_DLL_EXPORT)
+    target_link_libraries(mono-profiler-aot monosgen-shared eglib_objects log)
```
Comment (lambdageek) [inline src/mono/mono/profiler/CMakeLists.txt:34]: @directhex 

```suggestion
    target_link_libraries(mono-profiler-aot monosgen-shared eglib_objects log monoapi)
```

That should fix the broken builds.  (Fallout from #64569)

Comment (lambdageek) [reply src/mono/mono/profiler/CMakeLists.txt:34]: (And possibly the same thing for `mono-profiler-log`, above

**Review by jonathanpeppers — COMMENTED** [review]
So I manually tested the build artifact, and I'm seeing:
```
02-18 16:56:20.751 20237 20237 W monodroid: Initializing profiler with options: aot:port=9999,output=/data/user/0/com.mono.profiler.hellomaui/files/.__override__/profile.aotprofile
02-18 16:56:20.752 20237 20237 W monodroid: The 'aot' profiler wasn't found in the main executable nor could it be loaded from 'libmono-profiler-aot.so'.
```
It is inside the .apk:
```
lib\arm64-v8a\libmono-profiler-aot.so
```
I reverted to the old files I built myself, and those seem to work. It can load:
```
02-18 17:04:21.449 26961 26961 W monodroid: Looking for profiler init symbol 'mono_profiler_init_aot'? 0x7000c7855c
```

Is there enough changes in main, where a build from this PR wouldn't work alongside `release/6.0`?

**Comment by steveisok** [comment]
@jonathanpeppers is what you're trying to build easy so that we can test the same thing?

**Comment by steveisok** [comment]
> This symbol wasn't there before, would this cause an issue?

Yes, you're going to need a lib that is 1-1 with the runtime version you're using.  I can generate one from `release/6.0` that you could try.

**Review by jonathanpeppers — APPROVED** [review]
I manually tested a build based on `release/6.0` and it seems to work:

https://dev.azure.com/dnceng/internal/_build/results?buildId=1627363&view=results

```
  Read total 380759 bytes...
  Summary:
        Modules:         30
        Types:          966
        Methods:      3,969
  Going to write the profile to 'custom.aprof'
```

**Comment by steveisok** [comment]
Since the transport package is most useful in release/6.0, we shouldn't merge this to main just yet.

**Pull request closed without merging**